modules/src/mock/context.rs

```diff
@@ -268,10 +318,161 @@ impl MockContext {
+        self.port_capabilities.get(port_id).cloned()
+    }
+
+    fn autenthenticate(&self, _cap: &Capability, _port_id: &PortId) -> bool {
+        true
+    }
+}
+
+impl ChannelReader for MockContext {
+    fn channel_end(&self, pcid: &(PortId, ChannelId)) -> Option<ChannelEnd> {
+        self.channels.get(pcid).cloned()
+    }
+
+    fn connection_state(&self, cid: &ConnectionId) -> Option<ConnectionEnd> {
+        self.connections.get(cid).cloned()
+    }
+
+    fn connection_channels(&self, cid: &ConnectionId) -> Option<Vec<(PortId, ChannelId)>> {
+        self.connection_channels.get(cid).cloned()
+    }
+
+    fn channel_client_state(
+        &self,
+        port_channel_id: &(PortId, ChannelId),
+    ) -> Option<AnyClientState> {
+        let channel = self.channel_end(port_channel_id);
+        match channel {
+            Some(v) => {
+                let cid = v.connection_hops().clone()[0].clone();
+                let conn = self.connection_state(&cid);
+                match conn {
+                    Some(v) => ConnectionReader::client_state(self, &v.client_id().clone()),
+                    None => panic!(),
+                }
+            }
+            None => panic!(),
+        }
+    }
+
+    fn channel_consensus_state(
+        &self,
+        port_channel_id: &(PortId, ChannelId),
+        height: Height,
+    ) -> Option<AnyConsensusState> {
+        let channel = self.channel_end(port_channel_id);
+        match channel {
+            Some(v) => {
+                let cid = v.connection_hops().clone()[0].clone();
+
+                let conn = self.connection_state(&cid);
+                match conn {
+                    Some(v) => {
+                        ConnectionReader::client_consensus_state(
+                            self,
+                            &v.client_id().clone(),
+                            height,
+                        )
+                    }
+                    None => panic!(),
+                }
+            }
+            None => panic!(),
+        }
+    }
+
+    fn port_capability(&self, port_id: &PortId) -> Option<Capability> {
+        PortReader::lookup_module_by_port(self, port_id)
+    }
+
+    fn capability_authentification(&self, port_id: &PortId, cap: &Capability) -> bool {
+        PortReader::autenthenticate(self, cap, port_id)
+    }
+}
+
+impl ChannelKeeper for MockContext {
+    fn next_channel_id(&mut self) -> ChannelId {
+        let prefix = ChannelId::default().to_string();
+        let suffix = self.channel_ids_counter;
+        self.channel_ids_counter += 1;
+
+        ChannelId::from_str(format!("{}-{}", prefix, suffix).as_str()).unwrap()
+    }
+
+    fn store_channel(
+        &mut self,
+        port_channel_id: &(PortId, ChannelId),
+        channel_end: &ChannelEnd,
+    ) -> Result<(), ICS4Error> {
+        self.channels
+            .insert(port_channel_id.clone(), channel_end.clone());
+        Ok(())
+    }
+    fn store_nextsequence_send(
```

Comment:
A question of style (feel free to ignore). I'd rename these methods to be consistent with the fields `store_next_sequence_send`, `store_next_sequence_recv`, etc.

```suggestion
    fn store_next_sequence_send(
```